pdc/apps/common/tests.py

```diff
@@ -785,3 +785,40 @@ def test_delete_notification(self, mock_router):
                                  'nickname': 'Spot'
                              },
                          })])
+
+
+class TestNumericIDMixin(TestCase):
+    def setUp(self):
+        class Base(object):
+            def __init__(self, **kwargs):
+                self.kwargs = kwargs
+                self.get_object_called = False
+                self.get_queryset_called = False
+                self.queryset = mock.Mock()
+
+            def get_object(self):
+                self.get_object_called = True
+
+            def get_queryset(self):
+                self.get_queryset_called = True
+                return self.queryset
+
+        class DummyView(viewsets.NumericIDMixin, Base):
+            lookup_field = 'short'
+
+        self.View = DummyView
+
+    def test_retrieve_by_id(self):
+        view = self.View(short=1)
```

Comment:
Not sure why is this needed. Why not use everywhere `pk` with numeric value instead of `short`?

Reply:
In the tests this is basically directly translated into `self.kwargs`. In real viewsets DRF is extracting the values from url regex, and the field name is determined by the name in the regex, which always corresponds to `lookup_field`. For the simple cases of one field identifier this is always the name of the human readable id field. For example the regex for products would be `(?P<short>[^/.]+)`. The field will match either number or string.

For multi field views the regex is more complicated, for example release variants will have  `(?P<pk>\d+)|(?P<release__release_id>[^/]+)(?P<variant_uid>[^/]+)`, so `self.kwargs` will either have `pk` key or the two string identifiers.

Reply:
The regex from `NumericIDMixin` contains `?<pk>` so why doesn't `self.View(pk=1)` work? It's quite confusing that `short` gets translated to `pk`.

Reply:
It's actually also wrong. It does not work correctly with bulk operations on multi field views.